docs/proposals/scheduledjob.md

```diff
@@ -0,0 +1,132 @@
+    ObjectMeta
+
+    // Spec is a structure defining the expected behavior of a job, including the schedule.
+    Spec ScheduledJobSpec
+
+    // Status is a structure describing current status of a job.
+    Status ScheduledJobStatus
+}
+
+// ScheduledJobList is a collection of scheduled jobs.
+type ScheduledJobList struct {
+    TypeMeta
+    ListMeta
+
+    Items []ScheduledJob
+}
+```
+
+`ScheduledJobSpec` structure is defined to contain all the information how the actual
+job execution will look like, including the `JobSpec` from [Job Controller API](job.md)
+and the schedule in ISO8601 format.  This implies that each ScheduledJob execution
+will be created from the JobSpec actual at a point in time when the execution will
+be started.
+
+```go
+// ScheduledJobSpec describes how the job execution will look like and when it will actually run.
+type ScheduledJobSpec struct {
+
+    // JobSpec is a structure defining the expected behavior of a job.
+    JobSpec JobSpec
+
+    // Schedule contains the schedule in ISO8601 format, eg.
+    // - 2015-07-21T14:00:00Z - represents date and time in UTC
+    // - R/2015-07-21T14:00:00Z/P1D - represents endlessly repeating interval (1 day), starting from given date
+    Schedule string
+
+    // ConcurrentPolicy specifies how to treat concurrent executions of a Job.
+    ConcurrentPolicy ConcurrentPolicy
+
+    // BlockOnFailure suspends scheduling of next job runs after a failed one.
+    BlockOnFailure bool
+}
+
+// ConcurrentPolicy describes how the job will be handled.
+// Only one of the following concurrent policies may be specified.
+// If none of the following policies is specified, the default one
+// is AllowConcurrent.
+type ConcurrentPolicy string
+
+const (
+    // AllowConcurrent allows ScheduledJobs to run concurrently.
+    AllowConcurrent ConcurrentPolicy = "AllowConcurrent"
+
+    // SkipNextConcurrent forbids concurrent runs, skipping next run if previous
+    // hasn't finished yet.
+    SkipNextConcurrent ConcurrentPolicy = "SkipNextConcurrent"
+
+    // CancelPreviousConcurrent forbids concurrent runs, canceling currently
+    // running job and replacing it with a new one.
+    CancelPreviousConcurrent ConcurrentPolicy = "CancelPreviousConcurrent"
+)
+```
+
+`ScheduledJobStatus` structure is defined to contain some information for scheduled
+job executions (up to a limited amount, configured in master configuration).  The
+structure holds objects in three lists:
+* "Pending" list is used as a way to stack executions for job preventing concurrent
+runs, and also to trigger an execution for a job defined without time based scheduling;
+* "Running" list contains all actually running jobs with detailed information;
+* "Failed" list contains information about recently failed jobs.
+
+```go
+// ScheduledJobStatus represents the current state of a Job.
+type ScheduledJobStatus struct {
+
+    // PendingExecutions are the job runs pending for execution
+    PendingExecutions []Job
```

Comment:
Yeah, I'll address them as soon as we have consensus on Job API, which I'm hoping will happen withing next couple of days :)